### PR TITLE
Add loot table modification event

### LINKED ISSUE #42: Loot Table injecting
We need to be able to inject into vanilla (or other modded) loot tables using some sort of event or by offering some special JSON file syntax.

This would be used by most mods that add loot to stuff like dungeon chests, or that want to add to things like block drops (e.g. extra seeds from grass)

### PR DESCRIPTION
**Pull request by Juuxel**
Closes #42. It's a simple wrapper around the platform events.

(this is for 1.19 but should be the exact same when backporting)

### CODE REVIEW

File: common/src/main/java/dev/architectury/event/events/common/LootEvent.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    @FunctionalInterface
+    interface ModifyLootTable {
+        /**
+         * Modifies a loot table.
+         *
+         * @param lootTables the {@link LootTables} instance containing all loot tables
+         * @param id         the loot table ID
+         * @param context    the context used to modify the loot table
+         */
+        void modifyLootTable(LootTables lootTables, ResourceLocation id, LootTableModificationContext context);
```
Comment (shedaniel) [inline common/src/main/java/dev/architectury/event/events/common/LootEvent.java]: Would be nice as per discussed previously the `isBuiltin` param

Comment (Juuxel) [reply common/src/main/java/dev/architectury/event/events/common/LootEvent.java]: Yeah I've just forgotten about this pr, I'll fix it up today

Comment (Juuxel) [reply common/src/main/java/dev/architectury/event/events/common/LootEvent.java]: Okay done, also rebased onto master

**Review by shedaniel — APPROVED** [review]

**Review by MaxNeedsSnacks — APPROVED** [review]

**Comment by MaxNeedsSnacks** [comment]
Something to consider for the future: Implement non-builtin loot table modification on Forge, either through a mixin or by requesting (or PRing) this as a feature on their end

**Pull request merged**